kafka-ui-api/src/main/java/com/provectus/kafka/ui/emitter/MessageFilters.java

```diff
@@ -0,0 +1,88 @@
+package com.provectus.kafka.ui.emitter;
+
+import com.provectus.kafka.ui.exception.ValidationException;
+import com.provectus.kafka.ui.model.MessageFilterTypeDTO;
+import com.provectus.kafka.ui.model.TopicMessageDTO;
+import groovy.json.JsonSlurper;
+import java.util.function.Predicate;
+import javax.annotation.Nullable;
+import javax.script.CompiledScript;
+import javax.script.ScriptEngineManager;
+import javax.script.ScriptException;
+import org.apache.commons.lang3.StringUtils;
+import org.codehaus.groovy.jsr223.GroovyScriptEngineImpl;
+
+
+public class MessageFilters {
+
+  private static GroovyScriptEngineImpl GROOVY_ENGINE;
+
+  public static Predicate<TopicMessageDTO> createMsgFilter(String query, MessageFilterTypeDTO type) {
```

Comment:
Just a design idea but what if you would introduce a Interface `MessageFilter` with a filter method here ? Then you could have 2 impl. or 1 DefaultMessageFilter that takes a predicate when being created ?

Reply:
@jonasgeiregat I dont see any motivation to create more classes, if this interface just fit these needs